pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -257,49 +272,34 @@ func (admitter *VMsAdmitter) authorizeVirtualMachineSpec(ar *admissionv1.Admissi
 	var causes []metav1.StatusCause
 
 	for idx, dataVolume := range vm.Spec.DataVolumeTemplates {
-		cloneSource, err := typesutil.GetCloneSourceWithInformers(vm, &dataVolume.Spec, admitter.DataSourceInformer)
-		if err != nil {
-			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeUnexpectedServerResponse,
-				Message: err.Error(),
-				Field:   k8sfield.NewPath("spec", "dataVolumeTemplates").Index(idx).String(),
-			})
-
-			continue
-		}
-
-		if cloneSource == nil {
-			continue
-		}
-
-		if cloneSource.Namespace == "" {
-			cloneSource.Namespace = ar.Namespace
+		targetNamespace := vm.Namespace
+		if targetNamespace == "" {
+			targetNamespace = ar.Namespace
 		}
-
-		if cloneSource.Namespace == "" || cloneSource.Name == "" {
+		if dataVolume.Namespace != "" && dataVolume.Namespace != targetNamespace {
 			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeFieldValueNotFound,
-				Message: fmt.Sprintf("Clone source %s/%s invalid", cloneSource.Namespace, cloneSource.Name),
+				Type:    metav1.CauseTypeFieldValueInvalid,
+				Message: fmt.Sprintf("Embedded DataVolume namespace %s differs from VM namespace %s", dataVolume.Namespace, targetNamespace),
 				Field:   k8sfield.NewPath("spec", "dataVolumeTemplates").Index(idx).String(),
 			})
 
 			continue
 		}
-
-		targetNamespace := vm.Namespace
-		if targetNamespace == "" {
-			targetNamespace = ar.Namespace
-		}
-
-		serviceAccount := "default"
+		serviceAccountName := "default"
 		for _, vol := range vm.Spec.Template.Spec.Volumes {
 			if vol.ServiceAccount != nil {
-				serviceAccount = vol.ServiceAccount.ServiceAccountName
+				serviceAccountName = vol.ServiceAccount.ServiceAccountName
 			}
 		}
 
-		allowed, message, err := admitter.cloneAuthFunc(cloneSource.Namespace, cloneSource.Name, targetNamespace, serviceAccount)
-		if err != nil {
+		proxy := &authProxy{client: admitter.VirtClient, dataSourceInformer: admitter.DataSourceInformer}
+		dv := &cdiv1.DataVolume{
+			ObjectMeta: dataVolume.ObjectMeta,
+			Spec:       dataVolume.Spec,
+		}
+		dv.Namespace = targetNamespace
```

Comment:
Oh man, the the fact that you have to create a tmp DataVolume and explicitly set the namespace bums me out pretty bad.  Nothing we can do now but wish we thought of that earlier.